pandas/_libs/tslibs/src/datetime/np_datetime_strings.c

```diff
@@ -66,10 +66,19 @@ This file implements string parsing and creation for NumPy datetime.
  *
  * Returns 0 on success, -1 on failure.
  */
+
+int format_startswith(char ch, int format_len, char format, int exact) {
+    if ((format_len && format != ch) || (exact && !format_len)) {
```

Comment:
Nit: let's probably restore the comments?

Reply:
Also a minor. What C do you use, and with what optimization flags? Maybe we could also inline these checks.

Reply:
As in, `inline int format_startswith`?

I think this is where all the C compilation information is set: https://github.com/pandas-dev/pandas/blob/main/setup.py

Reply:
Yes, this is the one. Should work here unless it earlier than C99 is used afair.